### ISSUE TITLE
Uncaught Error: Invalid child node: undefined

### ISSUE THREAD

**Issue by BinaryMuse** [atom/incompatible-packages#21]
After building 1.17.0-dev-bb8db2acb, I found that `incompatible-packages` was showing the icon in the status bar; clicking on it leads to this exception.

**Atom**: 1.17.0-dev-bb8db2acb x64
**Electron**: 1.3.14
**OS**: Mac OS X 10.11.6
**Thrown From**: [incompatible-packages](https://github.com/atom/incompatible-packages) package 0.27.2

### Stack Trace

Uncaught Error: Invalid child node: undefined

```
At /Users/mtilley/github/atom/out/Atom.app/Contents/Resources/app/node_modules/etch/lib/dom.js:25

Error: Invalid child node: undefined
    at Object.dom (/Users/mtilley/github/atom/out/Atom.app/Contents/Resources/app/node_modules/etch/lib/dom.js:25:1)
    at /Users/mtilley/github/atom/out/Atom.app/Contents/Resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:207:1
    at Array.map (native)
    at IncompatiblePackagesComponent.renderIncompatibleModules (/Users/mtilley/github/atom/out/Atom.app/Contents/Resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:203:1)
    at IncompatiblePackagesComponent.renderIncompatiblePackage (/Users/mtilley/github/atom/out/Atom.app/Contents/Resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:156:1)
    at Array.map (native)
    at IncompatiblePackagesComponent.renderIncompatiblePackageList (/Users/mtilley/github/atom/out/Atom.app/Contents/Resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:132:1)
    at IncompatiblePackagesComponent.render (/Users/mtilley/github/atom/out/Atom.app/Contents/Resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:83:1)
    at updateSync (/Users/mtilley/github/atom/out/Atom.app/Contents/Resources/app/node_modules/etch/lib/component-helpers.js:99:1)
    at /Users/mtilley/github/atom/out/Atom.app/Contents/Resources/app/node_modules/etch/lib/component-helpers.js:72:1
    at DefaultScheduler.performUpdates (/Users/mtilley/github/atom/out/Atom.app/Contents/Resources/app/node_modules/etch/lib/default-scheduler.js:48:1)
```

### Commands

```
     -0:08.5.0 incompatible-packages:view (div.incompatible-packages-status.inline-block.text.text-error)
```

### Non-Core Packages

```
atom-beautify 0.29.17 
atom-spotify2 2.1.2 
atom-typescript 11.0.1 
autocomplete-emojis 2.5.0 
autocomplete-flow 1.6.0 
autocomplete-glsl 0.2.3 
autocomplete-go 1.1.1 
builder-go 1.0.1 
busy-signal 1.3.0 
chrome-color-picker 0.8.0 
elm-snippets 0.1.0 
file-icons 2.1.1 
fortune-background-tips 1.0.2 
git-pear 1.2.1 
go-config 1.2.3 
go-debug 0.1.22 
go-get 1.0.4 
go-plus 4.2.0 
godoc 0.4.0 
gofmt 1.1.8 
gometalinter-linter 1.1.1 
gorename 1.0.3 
hey-pane 0.2.1 
html-to-elm 0.2.0 
intentions 1.1.2 
jumpy 3.1.3 
language-babel 2.57.1 
language-elm 1.5.0 
language-graphql 0.8.0 
language-haskell 1.12.1 
language-pegjs 0.5.0 
linter 2.1.2 
linter-elm-make 0.23.2 
linter-eslint 8.1.4 
linter-flow 5.5.0 
linter-hlint 0.5.0 
linter-tslint 0.15.0 
linter-ui-default 1.2.1 
merge-conflicts 1.4.4 
narrow 0.39.0 
navigator-godef 1.0.3 
pigments 0.39.1 
react-es6-snippets 0.3.0 
relative-numbers 0.7.1 
smalls 0.3.0 
symbol-gen 1.3.1 
synesthete 0.0.5 
tester-go 1.0.4 
vim-mode-plus 0.84.1 
vim-mode-plus-ex-mode 0.7.1 
vim-mode-plus-subword-movement 0.2.2 
```

**Comment by tyrollins** [comment]
Same issue. Even after several fresh installs...This displays a red bug icon with "1" next to it. When  clicking said bug icon it brings a blank atom pane with with the tab bar title - "Incompatible Packages" & "Loading..." in the pane. spoiler - (Nothing ever loads)

Atom: 1.17.0-dev-96ddc516f
OS: macOS 10.12.4 (16E195)
Incompatable-Packages: 0.27.2

## Stack Trace

```
/Applications/Atom.app/Contents/Resources/app/node_modules/etch/lib/dom.js:25
Hide Stack Trace
Error: Invalid child node: undefined
    at Object.dom (/Applications/Atom.app/Contents/Resources/app/node_modules/etch/lib/dom.js:25:1)
    at /Applications/Atom.app/Contents/Resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:207:1
    at Array.map (native)
    at IncompatiblePackagesComponent.renderIncompatibleModules (/Applications/Atom.app/Contents/Resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:203:1)
    at IncompatiblePackagesComponent.renderIncompatiblePackage (/Applications/Atom.app/Contents/Resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:156:1)
    at Array.map (native)
    at IncompatiblePackagesComponent.renderIncompatiblePackageList (/Applications/Atom.app/Contents/Resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:132:1)
    at IncompatiblePackagesComponent.render (/Applications/Atom.app/Contents/Resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:83:1)
    at updateSync (/Applications/Atom.app/Contents/Resources/app/node_modules/etch/lib/component-helpers.js:99:1)
    at /Applications/Atom.app/Contents/Resources/app/node_modules/etch/lib/component-helpers.js:72:1
    at DefaultScheduler.performUpdates (/Applications/Atom.app/Contents/Resources/app/node_modules/etch/lib/default-scheduler.js:48:1)
```

## User Installed Packages

```
Remote-FTP 0.10.7
atom-beautify 0.29.17
autoprefixer 3.6.1
busy-signal 1.3.0
editorconfig 2.2.2
emmet 2.4.3
file-icons 2.1.2
highlight-selected 0.13.1
intentions 1.1.2
language-babel 2.57.2
language-haml 0.24.1
linter 2.1.2
linter-bootlint 1.0.4
linter-coffeelint 1.2.1
linter-eslint 8.1.4
linter-htmlhint 1.3.3
linter-php 1.3.2
linter-rubocop 1.1.1
linter-ui-default 1.2.2
multi-cursor 2.1.5
pigments 0.39.1
rspec 0.3.2
toggle-quotes 1.0.1
```
note: all "atom core packages" are active.

**Comment by tyrollins** [comment]
In my case, uninstalling Remote-FTP solved the issue.

**Comment by Jorge-Rodriguez** [comment]
:+1:  I find this issue on 1.16.0

**Comment by GrantRVD** [comment]
I have the same error after building from source. I haven't used Atom to edit anything yet, but I've installed several recommended packages by others. I don't have Remote-FTP installed like @tyrollins, so that isn't the source of my error. Note that I'm using Ubuntu, but my stack trace is identical to @BinaryMuse's. I'm using the most recent LTS of Node to build (v6.10.2).
`[Enter steps to reproduce:]

1. Launch Atom (from command-line or Launcher bar, both will have the error)
2. Try to show incompatible packages (e.g. click the red bug)
3. The incompatible packages tab will say 'Loading...' but never show anything

**Atom**: 1.18.0-dev-a73bb7e x64
**Electron**: 1.3.14
**OS**: Ubuntu 15.04
**Thrown From**: [incompatible-packages](https://github.com/atom/incompatible-packages) package 0.27.2


### Stack Trace

Uncaught Error: Invalid child node: undefined

```
At /usr/share/atom/resources/app/node_modules/etch/lib/dom.js:25

Error: Invalid child node: undefined
    at Object.dom (/usr/share/atom/resources/app/node_modules/etch/lib/dom.js:25:1)
    at /usr/share/atom/resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:207:1
    at Array.map (native)
    at IncompatiblePackagesComponent.renderIncompatibleModules (/usr/share/atom/resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:203:1)
    at IncompatiblePackagesComponent.renderIncompatiblePackage (/usr/share/atom/resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:156:1)
    at Array.map (native)
    at IncompatiblePackagesComponent.renderIncompatiblePackageList (/usr/share/atom/resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:132:1)
    at IncompatiblePackagesComponent.render (/usr/share/atom/resources/app/node_modules/incompatible-packages/lib/incompatible-packages-component.js:83:1)
    at updateSync (/usr/share/atom/resources/app/node_modules/etch/lib/component-helpers.js:99:1)
    at /usr/share/atom/resources/app/node_modules/etch/lib/component-helpers.js:72:1
    at DefaultScheduler.performUpdates (/usr/share/atom/resources/app/node_modules/etch/lib/default-scheduler.js:48:1)
```

### Commands

```
```

### Non-Core Packages

```
ariake-dark-syntax 0.1.3 
autocomplete-paths 1.0.5 
autocomplete-python 1.8.63 
build 0.67.0 
busy 0.7.0 
emmet 2.4.3 
git-plus 7.4.0 
git-plus-plus 1.1.0 
Hydrogen 1.11.1 
latex 0.43.0 
latex-completions 0.3.2 
markdown-pdf 1.5.0 
markdown-preview-enhanced 0.10.12 
markdown-table-formatter 2.9.0 
markdown-writer 2.6.4 
minimap 4.27.1 
platformio-ide-terminal 2.5.0 
```

**Comment by GrantRVD** [comment]
I should also point out that I looked at all the Non-Core Package lists above. The only one I share with @tyrollins is `emmet 2.4.3`, and I don't share any Non-Core Packages with @BinaryMuse, so this doesn't seem to be a problem resulting from package installation.

**Comment by pittix** [comment]
If it can be useful, there's my stacktrace with packages:
After updating atom via pacman,  the error I get is the same as reported by tyrollins. My atom and electron version are newer than the one reported by the other users.
**Atom**: 1.16.0 x64
**Electron**: 1.4.15
**OS**: "Arch Linux"
**Thrown From**: [incompatible-packages](https://github.com/atom/incompatible-packages) package 0.27.2


### Stack Trace

Uncaught Error: Invalid child node: undefined

```
At /usr/lib/atom/node_modules/incompatible-packages/node_modules/etch/lib/dom.js:25

Error: Invalid child node: undefined
    at Object.dom (/usr/lib/atom/node_modules/incompatible-packages/node_modules/etch/lib/dom.js:25:15)
    at /usr/lib/atom/node_modules/incompatible-packages/lib/incompatible-packages-component.js:203:31
    at Array.map (native)
    at IncompatiblePackagesComponent.renderIncompatibleModules (/usr/lib/atom/node_modules/incompatible-packages/lib/incompatible-packages-component.js:199:34)
    at IncompatiblePackagesComponent.renderIncompatiblePackage (/usr/lib/atom/node_modules/incompatible-packages/lib/incompatible-packages-component.js:152:63)
    at Array.map (native)
    at IncompatiblePackagesComponent.renderIncompatiblePackageList (/usr/lib/atom/node_modules/incompatible-packages/lib/incompatible-packages-component.js:128:35)
    at IncompatiblePackagesComponent.render (/usr/lib/atom/node_modules/incompatible-packages/lib/incompatible-packages-component.js:79:14)
    at updateSync (/usr/lib/atom/node_modules/incompatible-packages/node_modules/etch/lib/component-helpers.js:99:34)
    at /usr/lib/atom/node_modules/incompatible-packages/node_modules/etch/lib/component-helpers.js:72:7
    at DefaultScheduler.performUpdates (/usr/lib/atom/node_modules/incompatible-packages/node_modules/etch/lib/default-scheduler.js:48:34)
```

### Commands

```
```

### Non-Core Packages

```
atom-gdb 0.6.0 
autocomplete-clang 0.10.1 
busy-signal 1.3.0 
docblockr 0.9.4 
git-plus 7.4.0 
intentions 1.1.2 
language-generic-config 1.3.0 
language-matlab 0.2.1 
language-tex 0.4.0 
latex 0.43.0 
latexer 0.3.0 
linter 2.1.4 
linter-clang 4.1.1 
linter-spell 0.13.1 
linter-spell-latex 0.9.2 
linter-ui-default 1.2.3 
pdf-view 0.57.0 
terminal-fusion 2.4.4 
todo-show 1.11.0 
```

**Comment by pittix** [comment]
I have the same stacktrace as described in previous comments.
a temp fix is running in terminal the command 
```
apm install incompatible-packages@0.27.1
```
The problem is avoided and atom gives a deprecation warning, but all the environment (atom + other packages) work fine.

**Comment by edwinksl** [comment]
I have also encountered the same problem. In particular, the `latex` package doesn't work for me at all. Here are some details of my Atom Beta setup and OS:

```
$ apm-beta list -ib
auto-indent@0.5.0
autocomplete-python@1.8.63
busy-signal@1.3.0
file-icons@2.1.3
highlight-line@0.12.0
highlight-selected@0.13.1
hyperclick@0.0.40
intentions@1.1.2
language-cython@0.3.0
language-latex@1.0.0
language-markdown@0.21.0
language-matlab@0.2.1
language-restructuredtext@1.1.0
latex@0.43.0
latexer@0.3.0
linter@2.1.4
linter-chktex@1.3.0
linter-csslint@1.3.4
linter-flake8@2.1.5
linter-markdown@4.0.0
linter-matlab@1.1.0
linter-ui-default@1.2.3
markdown-scroll-sync@2.1.2
pdf-view@0.57.0
python-tools@0.6.9
todo-show@1.11.0
wordcount@2.10.4
```
```
$ lsb_release -a
LSB Version:	core-9.20160110ubuntu5-amd64:core-9.20160110ubuntu5-noarch:security-9.20160110ubuntu5-amd64:security-9.20160110ubuntu5-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 17.04
Release:	17.04
Codename:	zesty
```

**Comment by Dundee** [comment]
After installing incompatible-packages@0.27.1 the list of outdated packages is still not displayed and I get error:

**Atom**: 1.16.0 x64
**Electron**: 1.4.15
**OS**: linux 4.10.10-1-ARCH
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'text' of undefined

```
At /usr/lib/atom/node_modules/etch/lib/render.js:6

TypeError: Cannot read property 'text' of undefined
    at render (/usr/lib/atom/node_modules/etch/lib/render.js:6:18)
    at addChildren (/usr/lib/atom/node_modules/etch/lib/render.js:39:24)
    at render (/usr/lib/atom/node_modules/etch/lib/render.js:29:21)
    at addChildren (/usr/lib/atom/node_modules/etch/lib/render.js:39:24)
    at render (/usr/lib/atom/node_modules/etch/lib/render.js:29:21)
    at addChildren (/usr/lib/atom/node_modules/etch/lib/render.js:39:24)
    at render (/usr/lib/atom/node_modules/etch/lib/render.js:29:21)
    at addChildren (/usr/lib/atom/node_modules/etch/lib/render.js:39:24)
    at render (/usr/lib/atom/node_modules/etch/lib/render.js:29:21)
    at addChildren (/usr/lib/atom/node_modules/etch/lib/render.js:39:24)
    at render (/usr/lib/atom/node_modules/etch/lib/render.js:29:21)
    at updateChildren (/usr/lib/atom/node_modules/etch/lib/patch.js:94:36)
    at patch (/usr/lib/atom/node_modules/etch/lib/patch.js:15:9)
    at updateSync (/usr/lib/atom/node_modules/etch/lib/component-helpers.js:108:20)
    at /usr/lib/atom/node_modules/etch/lib/component-helpers.js:72:7
    at DefaultScheduler.performUpdates (/usr/lib/atom/node_modules/etch/lib/default-scheduler.js:48:34)
```

### Commands

```
     -0:34.9.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-always.compact-tab-bar.use-animations.compact-tree-view.theme-atom-material-syntax.theme-atom-material-ui.has-custom-icons)
     -0:26.3.0 incompatible-packages:view (div.incompatible-packages-status.inline-block.text.text-error)
```

### Non-Core Packages

```
atom-beautify 0.29.22 
atom-jinja2 0.6.0 
atom-material-syntax 1.0.2 
atom-material-ui 1.3.9 
atom-ternjs 0.18.2 
autocomplete-python 1.8.63 
busy-signal 1.3.0 
custom-title 1.0.1 
file-icons 2.1.3 
genesis-syntax 1.0.9 
genesis-ui 0.5.0 
git-blame 1.0.1 
goto-definition 1.3.2 
highlight-selected 0.13.1 
hyperclick 0.0.40 
insert-numerical-series 0.2.1 
intentions 1.1.2 
language-docker 1.1.7 
language-restructuredtext 1.1.0 
language-rust 0.4.10 
linter 2.1.4 
linter-docker 0.2.0 
linter-eslint 8.1.6 
linter-flake8 2.1.5 
linter-pycodestyle 2.0.2 
linter-pylint 2.0.1 
linter-rust 0.8.0 
linter-ui-default 1.2.3 
minimap 4.27.1 
project-manager 3.3.4 
python-isort 0.0.7 
qolor 0.4.1 
racer 0.20.0 
react 0.16.2 
sort-lines 0.14.0 
tree-ignore 0.6.2 
underscorify 0.1.0 
```

**Comment by Ceda-EI** [comment]
This certainly is related to external packages. I tried `atom --safe` and nothing shows up there.
Here is my list of non-community packages.
```
atom-beautify 0.29.22
atom-clock 0.1.7
atom-monokai 0.10.9
autocomplete-python 1.8.63
busy-signal 1.3.0
color-picker 2.2.5
file-icons 2.1.3
intentions 1.1.2
linter 2.1.4
linter-php 1.3.2
linter-pycodestyle 2.0.2
linter-ui-default 1.2.3
markdown-pdf 1.5.0
markdown-themeable-pdf 1.2.0
min-ui 0.0.4
minimap 4.27.1
minimap-find-and-replace 4.5.2
minimap-pigments 0.2.2
oceanic-next 1.0.0
outlander-syntax 2.0.0
outlander-ui 1.4.0
pdf-view 0.57.0
remote-edit2 3.0.0
seti-syntax 1.1.3
seti-ui 1.7.0
```
As a workaround to get rid of the bug in the status bar, I have temporarily disabled the `incompatible-packages` package.

**Comment by Jorge-Rodriguez** [comment]
Well, given that it shows up in all sorts of set ups with no common external packages, I'd sooner be inclined to think that atom's API for this is broken than blame external packages, especially when this is a problem that prevents users from seeing which packages may be at fault.

**Comment by 50Wliu** [comment]
Yeah.  By definition, this package is useless if no community packages are installed.

**Comment by ivlis** [comment]
@edwinksl have you fixed the problem with latex? I have exactly the same issue. Usually latex requires recompilation via the incompatible packages to work.

**Comment by edwinksl** [comment]
@ivlis I disabled `latex` (and also `latexer`) and I am using `latextools` for now.

**Comment by EdJoPaTo** [comment]
Looks like it is fixed by 77623c35e6bee47441067f210876540e4cca19c8.
I explicitly installed version 0.27.3 (`apm install incompatible-packages@0.27.3`) and the error is gone.
This is already included for the next release of atom: atom/atom@f4cfe7603fbe99c8f717eb92fdff20d33b3fd90a

**Comment by Jorge-Rodriguez** [comment]
I can confirm that the fix mentioned by @EdJoPaTo works for me on version 1.16.0

**Comment by Ben3eeE** [comment]
Thanks for the report everyone and apologies for the regression :bowing_man: We have cherry picked the commit so it will go out with the next hotfix release on beta. This is to ensure that the fix will be included once Atom 1.17 is released to stable.

Thanks for pointing out the commit that fixed this issue @EdJoPaTo :bowing_man:
